### ISSUE TITLE
Get-MgUser cannot return the list of transitive managers

### ISSUE THREAD

**Issue by alexandair** [microsoftgraph/msgraph-sdk-powershell#1792]
According to the [List manager](https://learn.microsoft.com/en-us/graph/api/user-list-manager) API doc, you need to use `Get-MgUser` cmdlet and specify `-UserId`, `-ExpandProperty`, `-CountVariable`, and `-ConsistencyLevel` parameters to get the list of transitive managers for a specific user.
But, the `-UserId`, `-CountVariable`, and `-ConsistencyLevel` parameters don't belong to the same parameter set.

Command mentioned in the docs is failing:
```powershell
Get-MgUser -UserId $userId -ExpandProperty "manager(`$levels=max;`$select=id,displayName)" -Property "id,displayName" -CountVariable CountVar -ConsistencyLevel eventual
```

How can we then use the `Get-MgUser` to get the management chain?

**Comment by peombwa** [comment]
Thanks for following up on this.

The HTTP example in the API reference is invalid as it goes against OData. We will follow up with the API owner to have it corrected. See the explanation at https://github.com/microsoftgraph/msgraph-sdk-powershell/issues/1749 for more details.

Marking as duplicate of https://github.com/microsoftgraph/msgraph-sdk-powershell/issues/1749.

**Comment by alexandair** [comment]
@peombwa If I understand you correctly, that HTTP example works by mistake. Is that right?

If you specify $expand and $levels=max, but don't specify $count, you get only the direct manager. Fetching other managers kind of silently fails.
ATM, you need to specify $count to get transitive managers.

If that's against OData, does that mean that we cannot use the single API call to get transitive managers?

**Comment by peombwa** [comment]
That's right! It's unfortunate that the HTTP example works despite the OData violation.

I've already informed the API owner on the invalid http example. I'll update the issue once I get a response on the next steps from API owner.

**Comment by Licantrop0** [comment]
@petrhollayms the API was fixed long ago, (adhering to OData standards, not requiring `$count`).

there are few problems in the generated cmdlet:
- ConsistencyLevel parameter is not available in that cmdlet due to missing annotations in the schema.
- The Manager property is not recognized as a proper User object (but just a generic DirectoryObject) and needs a cast.
- The manager chain is nested and not flat.

Here a script that solves the above:
```PowerShell
$user = Get-MgUser -UserId $userId -ExpandProperty "manager(`$levels=max;`$select=id,displayName)" -Property "id,displayName" -Headers @{ 'ConsistencyLevel' = 'Eventual' }

$CurrentLevel = 0
$ManagersChain = @([PSCustomObject]@{ Id = $user.Id; DisplayName = $user.DisplayName; Level = $CurrentLevel })

$currentManager = [Microsoft.Graph.PowerShell.Models.MicrosoftGraphUser]$user.Manager

while($currentManager.Id)
{
    $ManagersChain += [PSCustomObject]@{Id = $currentManager.Id; DisplayName = $currentManager.DisplayName; Level = ++$CurrentLevel }
    $currentManager = [Microsoft.Graph.PowerShell.Models.MicrosoftGraphUser]$currentManager.Manager
}

$ManagersChain
```

**Comment by alexandair** [comment]
@Licantrop0 That's a great script.
A minor correction could be to take into account that an organizational contact can also be a manager. Casting all managers to the User object works, but it's not accurate. Some of them could be orgContact.

I like the workaround with the -Headers parameter, but it would be great if the team could fix the cmdlet and support the -ConsistencyLevel parameter.

**Comment by Licantrop0** [comment]
@alexandair yes, the manager property can be either a `user` or an `orgContact​​`, and the script should manage this case. Also, the list (and levels) should be inverted so the CEO is always level 0 and the various reports on an increasing and consistent number.  
@SteveMutungi254: this could be a good script to add to [Entra PowerShell](https://github.com/microsoftgraph/entra-powershell)

For the faulty annotation, @timayabi2020 will take care (please link the tracking issue).

**Comment by timayabi2020** [comment]
Related metadata issue: https://github.com/microsoftgraph/msgraph-metadata/issues/762

**Comment by timayabi2020** [comment]
Missing annotation has been added.
